lib/puppet/provider/package/pkgng.rb

```diff
@@ -17,7 +17,7 @@ def self.get_query
   end
 
   def self.get_version_list
-    pkg(['version', '-voRL='])
+    @version_info_list ||= pkg(['version', '-voRL='])
```

Comment:
I know that in theory the agent is forked, but caching in providers still makes me kinda nervous.

It looks like this is only used in `self.instances`. I tink it would make more sense for that method to call `get_version_list` and then pass the list in to `get_latest_version`, instead of `get_latest_version` trying to query the version list each time it runs.

That way we only query the version list once a run still, but we don't risk stale data hanging around if the agent doesn't fork.